### PR TITLE
Fix a deadlock issue in ramfs

### PR DESCRIPTION
**Pull request by LclclcdIsACat**
related to #762

### CODE REVIEW

**Comment by liqinggd** [comment]
I think we should implement the `PageCacheBackend` for the inner of `RamInode`, which can fix the deadlock issue.

This fix cannot guarantee the concurrency of `resize`.May result in inconsistency between page cache size and file size.

File: kernel/aster-nix/src/fs/ramfs/fs.rs [hunk 1]
```diff
@@ -516,6 +516,8 @@ impl Inode for RamInode {
 
         let mut self_inode = self_inode.upgrade();
         self_inode.resize(new_size);
+        drop(self_inode);
+        let self_inode = self.0.read();
```
Comment (StevenJiang1110) [inline kernel/aster-nix/src/fs/ramfs/fs.rs:520]: This problem can also be resolved if implememting `downgrade` for `RwLockWriteGuard`, right? Like what [spin crate](https://docs.rs/spin/0.9.8/spin/rwlock/struct.RwLockWriteGuard.html#method.downgrade) does. By this way, we can avoid breaking atomicity.

**Comment by LclclcdIsACat** [comment]
> I think we should implement the `PageCacheBackend` for the metadata of `RamInode`, which can fix the deadlock issue.
> 
> This fix cannot guarantee the concurrency of `resize`.May result in inconsistency between page cache size and file size.

I agree with this point.

File: kernel/aster-nix/src/fs/ramfs/fs.rs (hunk 1 above)
Comment (liqinggd) [reply kernel/aster-nix/src/fs/ramfs/fs.rs:520]: I agree with the `downgrade` method

**Comment by tatetian** [comment]
@LclclcdIsACat Could you rebase this PR to the latest main so that we can be certain that it can pass the CI?

**Review by tatetian — APPROVED** [review]

**Comment by tatetian** [comment]
@LclclcdIsACat Just fix the minor doc issue and fix the conflicts. And we should be good.

**Pull request merged**